index.js

```diff
@@ -55,6 +55,12 @@ var TorControl = function TorControl(opts) {
         }
 
         this.connection = net.connect(params);
+        
+        //Handling connection errors
+        this.connection.once('error', function(err){
+            cb(new Error("Error connecting to control port: " + err));
```

Comment:
The callback is optional. Maybe you check if `cb` is `undefined` at the beginning and add a no-operation function to it, or you test (like used in the other parts) if `cb` exists before calling it.

```js
if (cb) {
  cb(new Error("Error connecting to control port: " + err));
}
```